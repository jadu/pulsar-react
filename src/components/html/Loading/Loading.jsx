import React from 'react';
import classnames from 'classnames';

export default class Loading extends React.Component {

  static defaultProps = {
    className: 'loading'
  };

  render() {
    let {
      block,
      children,
      className,
      large,
      ...props
    } = this.props;
    
    let variantClasses = classnames(className, {
      'loading--large': large
    });

    let spinner = (
      <span className={variantClasses} {...props}>
        <i>Loading...</i>
      </span>
    );

    let loadingBlock = block ? (
      <div className="loading-block">
        {spinner}
        {children &&
          <span className="loading-block__message">{children}</span>
        }
      </div>
    ) : spinner;

    return (
      <>{loadingBlock}</>
    );
  }
}