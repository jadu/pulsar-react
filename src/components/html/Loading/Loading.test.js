'use strict';

import React from 'react';
import { shallow, render } from 'enzyme';

import Loading from './Loading';

it('renders the base elements', () => {
  const tree = render(
    <Loading />
  );
  expect(tree).toMatchSnapshot();
});

it('renders the large variation', () => {
  const tree = render(
    <Loading large />
  );
  expect(tree).toMatchSnapshot();
});

it('renders the block variation', () => {
  const tree = render(
    <Loading block />
  );
  expect(tree).toMatchSnapshot();
});

it('allows a custom loading message with the block variation', () => {
  const tree = render(
    <Loading block>foo</Loading>
  );
  expect(tree).toMatchSnapshot();
});