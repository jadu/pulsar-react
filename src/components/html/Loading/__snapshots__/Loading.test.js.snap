// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows a custom loading message with the block variation 1`] = `
<div
  class="loading-block"
>
  <span
    class="loading"
  >
    <i>
      Loading...
    </i>
  </span>
  <span
    class="loading-block__message"
  >
    foo
  </span>
</div>
`;

exports[`renders the base elements 1`] = `
<span
  class="loading"
>
  <i>
    Loading...
  </i>
</span>
`;

exports[`renders the block variation 1`] = `
<div
  class="loading-block"
>
  <span
    class="loading"
  >
    <i>
      Loading...
    </i>
  </span>
</div>
`;

exports[`renders the large variation 1`] = `
<span
  class="loading loading--large"
>
  <i>
    Loading...
  </i>
</span>
`;
